File: ApolloCore/src/main/java/me/solarbam/apollo/apolloCore/utils/Valid.java
package me.solarbam.apollo.apolloCore.utils;

import me.solarbam.apollo.apolloCore.Common;

public final class Valid {

    private Valid() {
        // Utility class
    }

    public static void checkBoolean(boolean value, String message) {
        if (!value) {
            Common.throwError(new IllegalArgumentException(message), "A boolean check failed: " + message);
        }
    }

    public static void checkNotNull(Object value, String message) {
        if (value == null) {
            throw new IllegalArgumentException(message);
        }
    }

    public static void checkNotEmpty(String value) {
        checkNotNull(value, "Value cannot be null");
        if (value.isEmpty()) {
            throw new IllegalArgumentException("Value cannot be empty");
        }
    }
}

File: ApolloCore/src/main/kotlin/me/solar/apollo/apolloCore/utils/Valid.kt
package me.solar.apollo.apolloCore.utils

object Valid {
    fun checkBoolean(value: Boolean, message: String) {
        if (!value) {
            throw IllegalArgumentException(message)
        }
    }

    fun checkNotNull(value: Any?, message: String) {
        if (value == null) {
            throw IllegalArgumentException(message)
        }
    }

    fun checkNotEmpty(value: String) {
        checkNotNull(value, "Value cannot be null")
    }
}